internal/access/organization.go

```diff
@@ -35,16 +29,18 @@ func ListOrganizations(c *gin.Context, name string, pg *data.Pagination) ([]mode
 }
 
 func GetOrganization(c *gin.Context, id uid.ID) (*models.Organization, error) {
-	roles := []string{models.InfraSupportAdminRole}
-
-	// If the user is in the org, allow them to call this endpoint, otherwise they must be
-	// an InfraSupportAdmin.
-	db, err := hasAuthorization(c, id, isOrganizationSelf, roles...)
-	if err != nil {
-		return nil, HandleAuthErr(err, "organizations", "get", models.InfraSupportAdminRole)
+	rCtx := GetRequestContext(c)
+	if user := rCtx.Authenticated.User; user != nil && user.OrganizationID == id {
```

Comment:
I made one small change here from the implementation we had in `isOrganizationSelf`. It now checks the authorized `user.OrganizationID` instead of the authenticated `organization.ID`.

On unauthenticated routes the authenticated organization is still set, so the authenticated `user.OrganizationID` is a slightly safer pattern.  In this case it doesn't really matter because the route required authentication.